patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -292,21 +296,63 @@ public Recipe AddCondition(IEnumerable<Condition> conditions)
 		return this;
 	}
 
-	public Recipe DisableShimmer()
+	/// <summary>
+	/// Sets a condition delegate that will determine whether or not the recipe can be decrafted/changed in shimmer.. The condition can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="condition">The predicate delegate condition.</param>
+	/// <param name="description">A description of this condition. Use NetworkText.FromKey, or NetworkText.FromLiteral for this.</param>
+	public Recipe AddShimmerCondition(NetworkText description, Predicate<Recipe> condition) => AddShimmerCondition(new Condition(description, condition));
+
+	/// <summary>
+	/// Adds an array of conditions that will determine whether or not the recipe can be decrafted/changed in shimmer. The conditions can be unrelated to items or tiles (for example, biome or time).
+	/// </summary>
+	/// <param name="conditions">An array of conditions.</param>
+	public Recipe AddShimmerCondition(params Condition[] conditions) => AddShimmerCondition((IEnumerable<Condition>)conditions);
```

Comment:
I think we should replace `Shimmer` with `Decrafting` here... in the spirit of the original intent which is that there may be other things which decide to 'decraft' recipes, and shimmer doesn't only decraft. You already have `DecraftAvailable` above

Reply:
Damn, I had it right the second time